### ISSUE TITLE
x64 related crash on a boar texture (act_mutant_boar_u_bump)

### ISSUE THREAD

**Issue by Xottab-DUTY** [OpenXRay/xray-16#197]
Will add more info about the crash later.

Something wrong with the texture maybe?

**Comment by qweasdd136963** [comment]
act_mutant_boar_u_bump
![default](https://user-images.githubusercontent.com/20545492/39666867-123b76c4-50c4-11e8-90d6-c74616030073.png)

**Comment by revolucas** [comment]
I ran AXRToolset Thm validator:

`act_mutant_boar_u_bump has format DXT3 but dds has format DXT5`

There are other textures that have DXT3 in thm but are actually DXT5, so not sure this is the issue. But it's worth a try to fix thm and resave the image. Here:

https://drive.google.com/open?id=1mHB8SinX3xwfLdxms_Km0sVtZ5VVp4hB

**Comment by qweasdd136963** [comment]
@revolucas thanks! But it did not solve the problem. I still have this crash.

**Comment by NeoAnomaly** [comment]
@qweasdd136963 is there any additional information? the function return value for example

**Comment by qweasdd136963** [comment]
@NeoAnomaly function can not return anything. The game stops during the execution D3DX10CreateTextureFromMemory.

https://github.com/qweasdd136963/OXR_CoC/blob/f7abd2b5c7edde00f3f08d97d72a8e03faa9c8e4/src/Layers/xrRenderDX10/dx10Texture.cpp#L479

Crash occurs only if LoadInfo.Width <IMG.Width && LoadInfo.Height <IMG.Height.
The value of LoadInfo.Width and LoadInfo.Height is changed in the 464 line, in the Reduce function.

The first time I corrected it this way:
https://github.com/qweasdd136963/OXR_CoC/commit/f7abd2b5c7edde00f3f08d97d72a8e03faa9c8e4

But then I decided to redo it by taking a solution from the abramcumner repository:
https://github.com/qweasdd136963/OXR_CoC/commit/2a7344bf82f99c9699b001f447fdab0a00a54ee6#diff-b083c8845984cbb300f6966fc2b7f09bR21

**RUS:**
Функция не может ничего вернуть, т.к. вылет (или исключение, если запускаем через студию) происходит в процессе выполнения этой самой функции.

https://github.com/qweasdd136963/OXR_CoC/blob/f7abd2b5c7edde00f3f08d97d72a8e03faa9c8e4/src/Layers/xrRenderDX10/dx10Texture.cpp#L479

Вылет произойдет только если LoadInfo.Width < IMG.Width && LoadInfo.Height < IMG.Height.
Значение LoadInfo.Width и LoadInfo.Height изменяется в 464 строке, в функции Reduce.

Первый раз я исправил его так:
https://github.com/qweasdd136963/OXR_CoC/commit/f7abd2b5c7edde00f3f08d97d72a8e03faa9c8e4

Но затем решил переделать, взяв решение из репозитория abramcumner, которое возвращает установку FirstMipLevel:
https://github.com/qweasdd136963/OXR_CoC/commit/2a7344bf82f99c9699b001f447fdab0a00a54ee6#diff-b083c8845984cbb300f6966fc2b7f09bR21


P.S.
По сути проблема уже решена, но хотелось бы понять что было с текстурой.

**Comment by NeoAnomaly** [comment]
@qweasdd136963 such functions(dx api) internally well coded and tested and don't crash usually. So, probably your crash occured here:

```
HRESULT hr = expr;\
        if (!ignoreAlways && FAILED(hr))\
            xrDebug::Fail(ignoreAlways, DEBUG_INFO, #expr, hr, arg1);
```
the R_CHK macro is expanded to that, i.e. the engine log should contain the function return value.

May be you have a minidump of that crash?

**Comment by qweasdd136963** [comment]
No. It doesn't create minidump. There isn't anything in the log also.

**Comment by qweasdd136963** [comment]
I can provide only this.
![12](https://user-images.githubusercontent.com/20545492/40575095-b1725738-60f7-11e8-9daf-8159cc03bb38.png)

**Comment by NeoAnomaly** [comment]
@qweasdd136963 at first glance looks useful. I'll analyze it later

**added:**
Can you additionally provide the all field values of the IMG structure?

**Comment by qweasdd136963** [comment]
@NeoAnomaly, here.
![saf](https://user-images.githubusercontent.com/20545492/40577136-f5af4bf0-6119-11e8-861a-bd8df8dcef1f.png)

**Comment by qweasdd136963** [comment]
I've just checked another texture. That texture has same parameters, but all ok.. I think, the reason most likely needs to be found in the texture itself.

**RUS:**
Я только что проверил еще одну текстуру. Она имеет схожие с act_mutant_boar_u_bump параметры, но с ней все хорошо... Надо будет потом сравнить их настройки через редактор.
![h saf](https://user-images.githubusercontent.com/20545492/40577405-6ad54eb2-611e-11e8-9ffe-a513b0f61f04.png)

**Comment by Xottab-DUTY** [comment]
Fixed with 40fc8b473cf273bebfa221b2232cc692d6757b3b
